vcpkg/maintainers/functions/vcpkg_make_configure.md

```diff
@@ -0,0 +1,166 @@
+
+However, if you explicitly set `NO_CPPFLAGS` to true, you override this default behavior. Vcpkg will then leave the `-D` and `-isystem` flags untouched in their original variables (`CFLAGS` and `CXXFLAGS`). 
+
+If set, the `CPPFLAGS` environment variable, typically used for specifying C preprocessor flags, will not be automatically populated based on vcpkg's settings. This can be useful for projects that do not use standard environment variables or have specific preprocessor requirements.
+
+### ADD_BIN_TO_PATH
+
+Adds the configure dependent `(debug/)bin` directory to the system path. This is useful if configure builds and runs executables with vcpkg integrated dependencies.
+
+### NO_DEFAULT_OPTIONS
+
+When set, this flag disables the automatic inclusion of the following default configure options provided by `vcpkg_make_configure`. Use this flag if you need full control over the configure command line:
+
+General Defaults:
+* --disable-silent-rules - Makes the build process output more verbose.
+* --verbose - Enables verbose output from the configure script.
+
+Library Linkage:
+* For dynamic library builds: --enable-shared and --disable-static
+* For static library builds: --disable-shared and --enable-static
+
+Installation Directories:
+* Adjusts standard directories (bin, lib, share, etc.) to conform with vcpkg's layout, ensuring correct installation paths.
+
+Platform-Specific Tweaks:
+* On Windows: Ensures compatibility with Windows-specific filesystem and library behaviors.
+* On Unix-like systems: Assures the script accommodates the typical filesystem layout and toolchain behaviors.
+
+Removing these defaults gives you the responsibility to specify all necessary configuration options manually, ensuring the project is correctly configured for building with vcpkg.
+
+### NO_MSVC_FLAG_ESCAPING
+
+By default, escape characters (e.g., -Xcompiler, -Xlinker) are added before compiler and linker flags when using MSVC. This is done to ensure that these flags, which might contain spaces or other special characters, are correctly interpreted by the underlying MSVC tools (cl.exe, link.exe)
+
+When you set `NO_MSVC_FLAG_ESCAPING`, you tell vcpkg not to perform this automatic escaping. This can be useful in situations where:
+
+- You know your flags are formatted correctly: If you're confident that your compiler and linker flags don't contain problematic characters, you can avoid unnecessary escaping.
+- You're using custom build scripts: If you have custom build scripts or tools that require specific flag formatting, disabling automatic escaping might be necessary for compatibility.
+
+>[!NOTE]
+>If you disable the MSVC wrappers using `DISABLE_MSVC_WRAPPERS`, this option (`NO_MSVC_FLAG_ESCAPING`) has no effect because the escape characters are not added in the first place.
```

Comment:
Should this be reordered to be next to `DISABLE_MSVC_WRAPPERS` then?

Reply:
I think it would be good do add a table if possible describing the interactions between these two switches. Something like:

Input: `-Wall -O3 -isystem C:/headers/here -g /GL`

|                       | (default)                                    | `DISABLE_MSVC_WRAPPERS`                     |
|-----------------------|----------------------------------------------|-------------------------------------------|
| (default)             | `/W3 /O2 /IC:/headers/here /Z7 -Xcompiler /GL` | `-Wall -O3 -isystem C:/headers/here -g /GL` |
| `NO_MSVC_FLAG_ESCAPING` | `/W3 /O2 /IC:/headers/here /Z7 /GL`            | `-Wall -O3 -isystem C:/headers/here -g /GL` |